sonic-chassisd/scripts/chassisd

```diff
@@ -568,27 +600,23 @@ class ChassisdDaemon(daemon_base.DaemonBase):
             sys.exit(CHASSIS_LOAD_ERROR)
 
         # Check for valid slot numbers
-        my_slot = try_get(platform_chassis.get_my_slot,
-                                              default=INVALID_SLOT)
-        supervisor_slot = try_get(platform_chassis.get_supervisor_slot,
-                                                      default=INVALID_SLOT)
-        
+        my_slot = try_get(platform_chassis.get_my_slot, default=INVALID_SLOT)
+        supervisor_slot = try_get(platform_chassis.get_supervisor_slot, default=INVALID_SLOT)
+
         # Check if module list is populated
         self.module_updater = ModuleUpdater(SYSLOG_IDENTIFIER, platform_chassis, my_slot, supervisor_slot)
         self.module_updater.modules_num_update()
 
+        if not ModuleBase.MODULE_TYPE_DPU and \
+           not ModuleBase.MODULE_TYPE_SWITCH:
+            if ((self.module_updater.my_slot == INVALID_SLOT) or
+                    (self.module_updater.supervisor_slot == INVALID_SLOT)):
+                self.log_error("Chassisd not supported for this platform")
+                sys.exit(CHASSIS_NOT_SUPPORTED)
 
-        if ((self.module_updater.my_slot == INVALID_SLOT) or
-                (self.module_updater.supervisor_slot == INVALID_SLOT)):
-            self.log_error("Chassisd not supported for this platform")
-            sys.exit(CHASSIS_NOT_SUPPORTED)
-
-        # Start configuration manager task on supervisor module
-        if self.module_updater.supervisor_slot == self.module_updater.my_slot:
-            config_manager = ConfigManagerTask()
-            config_manager.task_run()
-        else:
-            config_manager = None
+        config_manager = ConfigManagerTask()
```

Comment:
ConfigManagerTask is derived from the ProcessTaskBase class. This class runs the task_worker in a separate process. Why is the process replaced with the thread? In Python due to the GIL limitation, it is preferable to use processes instead of threads.

Reply:
This is also not required.  Reverted the change.